maven-xml/src/main/java/org/apache/maven/xml/sax/filter/AbstractEventXMLFilter.java

```diff
@@ -0,0 +1,289 @@
+package org.apache.maven.xml.sax.filter;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+import org.apache.maven.xml.sax.SAXEvent;
+import org.apache.maven.xml.sax.SAXEventFactory;
+import org.xml.sax.Attributes;
+import org.xml.sax.Locator;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+import org.xml.sax.ext.LexicalHandler;
+
+/**
+ * Builds up a list of SAXEvents, which will be executed with {@link #executeEvents()}
+ * 
+ * @author Robert Scholte
+ * @since 3.7.0
+ */
+abstract class AbstractEventXMLFilter extends AbstractSAXFilter
+{
+    private Queue<SAXEvent> saxEvents = new ArrayDeque<>();
+    
+    private SAXEventFactory eventFactory;
+    
+    // characters BEFORE startElement must get state of startingElement
+    // this way removing based on state keeps correct formatting
+    private SAXEvent characters;
+    
+    private boolean lockCharacters = false;
+    
+    protected abstract boolean isParsing();
+    
+    protected abstract String getState();
+    
+    protected boolean acceptEvent( String state )
+    {
+        return true;
+    }
+    
+    AbstractEventXMLFilter()
+    {
+        super();
```

Comment:
Perhaps this [explicit] call, and few more to `super()`, could be removed.

Reply:
This is all to prevent confusion. This is not just the default constructor, there's a second one. Empty constructor often suggest generated and redundant, hence I'll keep the `super`-calls.